exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/xsort/TestSimpleExternalSort.java

```diff
@@ -22,220 +22,167 @@
 import org.junit.Ignore;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.rules.TestRule;
 
-import com.google.common.base.Charsets;
-import com.google.common.io.Files;
-
-@Ignore
-public class TestSimpleExternalSort extends BaseTestQuery {
+public class TestSimpleExternalSort extends DrillTest {
   static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(TestSimpleExternalSort.class);
-  DrillConfig c = DrillConfig.create();
-
 
   @Rule public final TestRule TIMEOUT = TestTools.getTimeoutRule(80000);
 
   @Test
-  public void mergeSortWithSv2() throws Exception {
-    List<QueryDataBatch> results = testPhysicalFromFileWithResults("xsort/one_key_sort_descending_sv2.json");
-    int count = 0;
-    for(QueryDataBatch b : results) {
-      if (b.getHeader().getRowCount() != 0) {
-        count += b.getHeader().getRowCount();
-      }
-    }
-    assertEquals(500000, count);
-
-    long previousBigInt = Long.MAX_VALUE;
-
-    int recordCount = 0;
-    int batchCount = 0;
-
-    for (QueryDataBatch b : results) {
-      if (b.getHeader().getRowCount() == 0) {
-        break;
-      }
-      batchCount++;
-      RecordBatchLoader loader = new RecordBatchLoader(allocator);
-      loader.load(b.getHeader().getDef(),b.getData());
-      BigIntVector c1 = (BigIntVector) loader.getValueAccessorById(BigIntVector.class,
-              loader.getValueVectorId(new SchemaPath("blue", ExpressionPosition.UNKNOWN)).getFieldIds()).getValueVector();
-
-
-      BigIntVector.Accessor a1 = c1.getAccessor();
+  public void mergeSortWithSv2Legacy() throws Exception {
+    mergeSortWithSv2(true);
+  }
 
-      for (int i =0; i < c1.getAccessor().getValueCount(); i++) {
-        recordCount++;
-        assertTrue(String.format("%d > %d", previousBigInt, a1.get(i)), previousBigInt >= a1.get(i));
-        previousBigInt = a1.get(i);
-      }
-      loader.clear();
-      b.release();
+  /**
+   * Tests the external sort using an in-memory sort. Relies on default memory
+   * settings to be large enough to do the in-memory sort (there is,
+   * unfortunately, no way to double-check that no spilling was done.)
+   * This must be checked manually by setting a breakpoint in the in-memory
+   * sort routine.
+   *
+   * @param testLegacy
+   * @throws Exception
+   */
+
+  private void mergeSortWithSv2(boolean testLegacy) throws Exception {
+    try (ClusterFixture cluster = ClusterFixture.standardCluster( );
+         ClientFixture client = cluster.clientFixture()) {
+      chooseImpl(client, testLegacy);
+      List<QueryDataBatch> results = client.queryBuilder().physicalResource("xsort/one_key_sort_descending_sv2.json").results();
+      assertEquals(500000, client.countResults( results ));
+      validateResults(client.allocator(), results);
     }
+  }
 
-    System.out.println(String.format("Sorted %,d records in %d batches.", recordCount, batchCount));
+  private void chooseImpl(ClientFixture client, boolean testLegacy) throws Exception {
```

Comment:
Will the implementation for this be added once the new sort mechanism is checked in ?